terraform-mode.el

```diff
@@ -97,8 +97,8 @@
 
   ;; imenu
   (setq imenu-generic-expression
-        '(("resource" "^resource\\s-+\"[^\"]+\"\\s-+\"\\([^\"]+\\)\"" 1)
-          ("data" "^data\\s-+\"[^\"]+\"\\s-+\"\\([^\"]+\\)\"" 1)
+        '(("resource" "^resource\\s-+\\(\"[^\"]+\"\\s-+\"[^\"]+\"\\)" 1)
+          ("data" "^data\\s-+\\(\"[^\"]+\"\\s-+\"[^\"]+\"\\)" 1)
```

Comment:
This will only find resource types and names that are within quotation marks.  

The following block defiitions are all valid:
```
resource foo bar {}
resource "foo" "bar" {}

data foo bar {}
data "foo" "bar"{}
```

In addition it would be nice if these were presented using the [rx macro](https://www.gnu.org/software/emacs/manual//html_node/elisp/Rx-Constructs.html#Rx-Constructs)

Reply:
That is interesting. I never used that syntax and didn't know that it was supported. I will make the requested changes as soon as I can.

Reply:
I pushed another commit that replaces strings with the usage of the rx macro.